kobo/worker/logger.py

```diff
@@ -71,6 +71,10 @@ def stop(self):
 
 class LoggingIO(object):
     """StringIO wrapper that also writes all data to a logging thread."""
+    __slots__ = (
```

Comment:
See the commit message.  The code was originally there.  At some point, the code was removed without any explanation.  Removing the code caused a serious regression in Covscan.  I was able to find the cause of the regression by bisecting the changes and isolating a minimal working patch out of the (reverted) first bad commit.

I am not able to explain why the partial revert of 290bda401907a003686646db49ffed2339a24521 fixes the bug because I am not a python programmer and I have no idea what slots are.  But making Covscan work again is a good enough reason to push it, given the fact that there are now known downsides of the proposed patch.